src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,390 @@
+        return this.getBucketId(args, planId);
+      })
+      .then(bucketId => {
+        args.options.bucketId = bucketId;
+        return this.generateUserAssignments(args);
+      })
+      .then(assignments => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/tasks`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none'
+          },
+          responseType: 'json',
+          data: {
+            planId: args.options.planId,
+            bucketId: args.options.bucketId,
+            title: args.options.title,
+            startDateTime: args.options.startDateTime,
+            dueDateTime: args.options.dueDateTime,
+            percentComplete: args.options.percentComplete,
+            assignments: assignments,
+            orderHint: args.options.orderHint
+          }
+        };
+
+        return request.post(requestOptions) as PlannerTask;
+      })
+      .then(newTask => { 
+        const taskId = newTask.id as string;
+  
+        if (args.options.description) {
+          return this.getTaskDetailsEtag(taskId)
+            .then(etag => {
+              const requestOptionsTaskDetails: any = {
+                url: `${this.resource}/v1.0/planner/tasks/${taskId}/details`,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none',
+                  'If-Match': etag,
+                  'Prefer': 'return=representation'
+                },
+                responseType: 'json',
+                data: {
+                  description: args.options.description
+                }
+              };
+      
+              return request.patch(requestOptionsTaskDetails);              
+            })
+            .then(taskDetails => {
+              return { ...newTask, ...taskDetails as PlannerTaskDetails };
+            });
+        } 
+        else {
+          return newTask;
+        }  
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(taskId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
+        }
+
+        return Promise.resolve(etag);
+      });
+  }
+
+  private generateUserAssignments(args: CommandArgs): Promise<{ [userId: string]: PlannerAssignment; }> {
+    const assignments: { [userId: string]: PlannerAssignment; } = {};
+
+    if (args.options.assignedToUserNames) {
```

Comment:
Let's simplify this block so that we don't need to call `Promise.resolve` 3 times. What if we moved the logic of retrieving userIds to the `getUserIds` function that either returns an early Promise.resolve with the IDs specified in the options or resolves names to IDs from the API when names were specified.